.github/workflows/wsl-exec.yaml

```diff
@@ -0,0 +1,196 @@
+          az_vm_username: ${{ secrets.az_vm_username }}
+          az_vm_password: ${{ secrets.az_vm_password }}
+
+  wsl-setup-and-script-execution:
+    strategy:
+      max-parallel: 1
+    needs: start-vm
+    runs-on: self-hosted
+    env:
+      LAUNCHER: "ubuntupreview.exe"
+      DISTRO: "Ubuntu-Preview"
+      APPX: "Ubuntu (Preview)"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install the certificate
+        shell: powershell
+        run: |
+          New-Item -ItemType directory -Path certificate
+          Set-Content -Path certificate\certificate.txt -Value '${{ secrets.certificate }}'
+          certutil -decode certificate\certificate.txt certificate\certificate.pfx
+
+          $pwd = ConvertTo-SecureString  '${{ secrets.certificate_password }}' -AsPlainText -Force
+          Import-PfxCertificate -Password $pwd -CertStoreLocation Cert:LocalMachine\Trust -FilePath certificate\certificate.pfx
+          Import-PfxCertificate -Password $pwd -CertStoreLocation Cert:CurrentUser\My -FilePath certificate\certificate.pfx
+      - name: Run
+        uses: ./.github/actions/wsl_enabled_powershell/
+        with:
+          exec: |
+            Write-Output '::group::Installing dependencies'
+            # Install or update WSL
+            winget install 'Windows Subsystem for Linux' --accept-source-agreements --accept-package-agreements --silent
+
+            # Install or update Ubuntu (Preview)
+            winget install '${{ env.APPX }}'  --accept-source-agreements  --accept-package-agreements --silent
+            Write-Output '::endgroup::'
+
+            Write-Output '::group::Setting up WSL'
+            $env:WSL_UTF8=1
+
+            # Unregister any pre-exising distro
+            wsl.exe --unregister ${{ env.DISTRO }} > NULL
+
+            # Register a new instance
+            & ${{ env.LAUNCHER }} install --root --ui=none
+
+            # Disable interop and reboot
+            & ${{ env.LAUNCHER }} run @'
+            cat << EOF >> /etc/wsl.conf
+
+            [interop]
+            enabled=false
+            appendWindowsPath=false
+
+            [automount]
+            enabled=false
+
+            EOF
+            '@
+
+            wsl --terminate ${{ env.DISTRO }}
+            Write-Output '::endgroup::'
+
+            Write-Output '::group::Cloning repo'
+            # Clone repo
+            If( ${{ inputs.repo }} -ne "REPO_NOT_PROVIDED"  ) {
+              & ${{ env.LAUNCHER }} run @'
+              git clone ${{ inputs.repo }} action_working_dir
+              cd action_working_dir
+              git checkout ${{ inputs.checkout }}
+            '@
+            } else {
+              Write-Output "Repo not provided, skipping"
+               & ${{ env.LAUNCHER }} run mkdir action_working_dir
+            }
+            Write-Output '::endgroup::'
+
+            Write-Output '::group::Storing script into WSL'
+            $tmpdir="\\wsl.localhost\${env:distro}\tmp\"
```

Comment:
Executing from `/tmp` can easily become a security concern, specially because it's easy to conflict with files generated by other programs. I'd recommend create a temporary directory and file with `mktemp` inside Linux and use that instead of `/tmp/script`.

Reply:
Makes sense, however `mktemp` won't do it because it does not create real directories, or at least Windows thinks they are files rather than directories.

Since the WSL instance is volatile anyways, I can just use some random long name for the folder.

Reply:
`/tmp/github_${{ github.run_id }}` should be good enough